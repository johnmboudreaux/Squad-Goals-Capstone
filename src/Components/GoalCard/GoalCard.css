.squads-body {
  text-align: center;
  background-color: #0e5bd7;
  box-shadow: 2px 2px 4px #282828;
  border-radius: 5px;
  color: #d2d3d3;
  height: 50vw;
  width: 50vw;
  margin: 10px auto 10px auto;
  opacity: .95; }

.squads-body-title {
  font-weight: lighter;
  font-size: 1.5em;
  margin: 10px 0 15px 0; }

.squads-body-info input {
  background-color: #22affe;
  border: none;
  border-radius: 3px; }
