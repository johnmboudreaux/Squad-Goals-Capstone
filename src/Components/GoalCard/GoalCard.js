import React from 'react';
import PropTypes from 'prop-types';

const GoalCard = ({ goal }) => {
  return (
    <tr className="goal-row">
      <td>{goal.title}</td>
      <td>{goal.goal_time}</td>
      <td>{goal.goal_points}</td>
      <td><input type="button" value="Leave" /></td>
    </tr>
  );
};

GoalCard.propTypes = {
  goal: PropTypes.object,
};

export default GoalCard;
