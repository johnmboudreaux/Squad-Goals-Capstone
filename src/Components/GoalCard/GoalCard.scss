@import '../../variables.scss';

.squads-body {
  text-align: center;
  background-color: $main;
  box-shadow: 2px 2px 4px $black;
  border-radius: 5px;
  color: $white;
  width: 60vw;
  margin: 10px auto 10px auto;
  opacity: .95;
}

.squads-body-title {
  font-weight: lighter;
  font-size: 1.5em;
  margin: 10px 0 15px 0;
}

// .squads-body-info {
//   align-items: center;
//   display: flex;
//   flex-direction: row;
//   justify-content: space-between;
//   font-size: .8em;
//   margin: 10px;
// }

.squads-body-info input {
  background-color: $main-light;
  border: none;
  border-radius: 3px;
}
