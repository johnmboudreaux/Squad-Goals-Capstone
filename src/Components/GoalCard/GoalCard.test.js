import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import GoalCard from './GoalCard';
import goal from '../../../data/test/goals_data_test';

describe('MainApp snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<GoalCard
      goal={goal[0]}
    />);

    expect(wrapper).toMatchSnapshot();
  });


  it('renders without crashing', () => {
    const tbody = document.createElement('tbody');
    ReactDOM.render(<GoalCard
      goal={goal[0]}
    />, tbody);
  });
});
