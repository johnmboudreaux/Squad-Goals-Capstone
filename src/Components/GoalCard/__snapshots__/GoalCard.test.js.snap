// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainApp snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoalCard
    goal={
      Object {
        "conversation_id": 1,
        "creator_id": 1,
        "description": "to rule them all",
        "goal_points": 2300,
        "goal_time": "2018-01-12 15:00:00",
        "id": 1,
        "title": "chess tourney",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td>
          <a
            href={
              Object {
                "pathname": "/Goal/1",
                "state": Object {
                  "userId": undefined,
                },
              }
            }
          >
            chess tourney
          </a>
        </td>,
        <td>
          2018-01-12 15:00:00
        </td>,
        <td>
          2300
        </td>,
      ],
      "className": "goal-row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href={
              Object {
                "pathname": "/Goal/1",
                "state": Object {
                  "userId": undefined,
                },
              }
            }
          >
            chess tourney
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "chess tourney",
            "href": Object {
              "pathname": "/Goal/1",
              "state": Object {
                "userId": undefined,
              },
            },
          },
          "ref": null,
          "rendered": "chess tourney",
          "type": "a",
        },
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "2018-01-12 15:00:00",
        },
        "ref": null,
        "rendered": "2018-01-12 15:00:00",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 2300,
        },
        "ref": null,
        "rendered": 2300,
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td>
            <a
              href={
                Object {
                  "pathname": "/Goal/1",
                  "state": Object {
                    "userId": undefined,
                  },
                }
              }
            >
              chess tourney
            </a>
          </td>,
          <td>
            2018-01-12 15:00:00
          </td>,
          <td>
            2300
          </td>,
        ],
        "className": "goal-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href={
                Object {
                  "pathname": "/Goal/1",
                  "state": Object {
                    "userId": undefined,
                  },
                }
              }
            >
              chess tourney
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "chess tourney",
              "href": Object {
                "pathname": "/Goal/1",
                "state": Object {
                  "userId": undefined,
                },
              },
            },
            "ref": null,
            "rendered": "chess tourney",
            "type": "a",
          },
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018-01-12 15:00:00",
          },
          "ref": null,
          "rendered": "2018-01-12 15:00:00",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2300,
          },
          "ref": null,
          "rendered": 2300,
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
